### PR TITLE
*: add builtin function SIGN

### PR DESCRIPTION
**Pull request by zyguan**
A function wanted by #2420.

### CODE REVIEW

File: expression/builtin_math_test.go [hunk 1]
```diff
@@ -231,3 +231,29 @@ func (s *testEvaluatorSuite) TestConv(c *C) {
...
+	}{
+		{nil, nil},
+		{1, 1},
+		{0, 0},
+		{-1, -1},
+		{0.4, 1},
+		{-0.4, -1},
+		{"1", 1},
+		{"-1", -1},
+		{"1a", 1},
```
Comment (XuHuaiyu) [inline expression/builtin_math_test.go]: You mean `0x1a` and `-0x1a` here?
Which type will `1a` be treated as in `types.MakeDaums`?

File: expression/builtin_math.go [hunk 2]
```diff
@@ -516,6 +519,42 @@ func builtinCRC32(args []types.Datum, ctx context.Context) (d types.Datum, err e
...
+
+// See http://dev.mysql.com/doc/refman/5.7/en/mathematical-functions.html#function_sign
+func builtinSign(args []types.Datum, ctx context.Context) (d types.Datum, err error) {
+	if args[0].IsNull() {
+		return d, nil
+	}
+	cmp, err := args[0].CompareDatum(ctx.GetSessionVars().StmtCtx, types.NewIntDatum(0))
+	switch {
+	case err == nil:
+	case terror.ErrorEqual(err, types.ErrTruncated):
```
Comment (tiancaiamao) [inline expression/builtin_math.go]: Truncate error can't be simply ignored, it's a warning in a select context, but an error in a insert context:

```
mysql> insert into t values (sign('1a'));
ERROR 1292 (22007): Truncated incorrect DOUBLE value: '1a'

mysql> select sign('1a');
+------------+
| sign('1a') |
+------------+
|          1 |
+------------+
1 row in set, 1 warning (0.00 sec)
```

Comment (zyguan) [inline expression/builtin_math.go]: Ah! I hadn't considered that. Thanks for pointing it out, @tiancaiamao .

**Comment by zyguan** [comment]
Hi, @tiancaiamao , PTAL, is [this](https://github.com/pingcap/tidb/pull/2427/commits/403754927ee71436d5d4abe7dbea767767b42bdc) ok?

**Comment by shenli** [comment]
LGTM  @tiancaiamao @coocood PTAL 
@zyguan Good Job!

**Comment by tiancaiamao** [comment]
LGTM
please resolve conflicts.

**Comment by zyguan** [comment]
Thanks @hanfei1991 ! I saw the second travis check was hanging a few days ago. However, I'm a little busy recently :(
I meant to merge the master to trigger the CI tonight but found that you've already done that.

**Pull request merged**